### ISSUE TITLE
lx-zone's on platfoms images after 20151126T062747Z can not use useradd , vipw , vigr

### ISSUE THREAD

**Issue by failedrequest** [joyent/illumos-joyent#96]
LX-Zones Running on platforms after 20151126T062747Z can not create/modify users, modify their shadow file, or group files.

For example running the lx-centos-6 image from  20150811 on 20151126T062747Z I can correctly run useradd and vipw.

On 20160108T173524Z useradd , groupadd, vipw all generate a corefile when run. 

After conferring with rzezeski we suspect this may be related to a change in how vfork works as denoted in this commit. 
https://github.com/joyent/illumos-joyent/commit/93c2b12b16d218a886047b7927b0b040473b2a42

Strace'ng the affected binaries produces the following hang that pointed ryan to thing it was due to vfork.

The strace is here http://pastebin.com/dkYtKKiZ

**Comment by pfmooney** [comment]
When that specific `vfork()` change was made, a bug was introduced which prevented ptrace consumers (read: strace and co.) from being able to properly follow `vfork()`.  The ptrace issue was fixed in 7db52ee64189522d4c1939f0b7723c3d91449e16.

I'm investigating the useradd/etc segfaults now.

**Comment by pfmooney** [comment]
This is an strace capture of the described segfault behavior:
```
unlink("/etc/gshadow.lock")             = 0
vfork(Process 136034 attached
 <unfinished ...>
[pid 136034] execve("/usr/sbin/nscd", ["/usr/sbin/nscd", "nscd", "-i", "passwd"], []) = -1 ENOENT (No such file or directory)
[pid 136034] exit_group(127)            = ?
[pid 136033] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=136034, si_status=127, si_utime=0, si_stime=0} ---
[pid 136033] <... vfork resumed> )      = 136034
[pid 136033] wait4(136034, [{WIFEXITED(s) && WEXITSTATUS(s) == 127}], 0, NULL) = 136034
[pid 136033] vfork(Process 136035 attached
 <unfinished ...>
[pid 136035] execve("/usr/sbin/nscd", ["/usr/sbin/nscd", "nscd", "-i", "group"], []) = -1 ENOENT (No such file or directory)
[pid 136035] exit_group(127)            = ?
[pid 136033] <... vfork resumed> )      = 136035
[pid 136033] --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_ACCERR, si_addr=0x7fffffeff290} ---
[pid 136033] +++ killed by SIGSEGV (core dumped) +++
Process 136034 detached
Process 136035 detached
Segmentation fault (core dumped)
```

**Comment by pfmooney** [comment]
Open as [OS-5124](https://smartos.org/bugview/OS-5124)

**Comment by pfmooney** [comment]
This should be resolved in 67304e3c6d5fac4da34a729495275ca86ab1ad1e.

Thanks for reporting the issue.
